src/libexpr/primops/fetchTree.cc

```diff
@@ -74,7 +74,7 @@ std::string fixURI(std::string uri, EvalState & state, const std::string & defau
 
 std::string fixURIForGit(std::string uri, EvalState & state)
 {
-    static std::regex scp_uri("([^/].*)@(.*):(.*)");
+    static std::regex scp_uri("^([^/]*)@(.*):(.*)");
```

Comment:
Note that a ^ at the start should be unnecessary since `regex_match` already must match the entire input.

Reply:
Thanks, works.